show/bgp_common.py

```diff
@@ -0,0 +1,420 @@
+                ns_l.append(name_s)
+            asic_cnt = len(ns_l)
+            if asic_cnt > 1 and display == constants.DISPLAY_ALL:
+                print_ns_str = True
+        except ValueError:
+            print "namespace '{}' is not valid. valid name spaces are:\n{}".format(namespace, multi_asic_util.multi_asic_ns_choices())
+            return
+    else:
+        if namespace is not None:
+            print "namespace option is not applicable for non-multi-asic platform"
+            return
+        if display is not None:
+            print "display option is not applicable for non-multi-asic platform"
+            return
+        ns_l.append(0)
+
+    # build the filter set only if necessary
+    if filter_back_end:
+        back_end_intf_set = multi_asic.get_back_end_interface_set()
+    else:
+        back_end_intf_set = None
+    # get all the other arguments except json that needs to be the last argument of the cmd if present
+    for arg in args:
+        arg_strg += str(arg) + " "
+        if str(arg) == "json":
+            found_json = 1
+        else:
+            try:
+                filter_by_ip = ipaddress.ip_network(arg)
+            except ValueError:
+                # Not ip address just ignore it
+                pass
+    if not found_json:
+        arg_strg += "json"
+    combined_route = {}
+    for ns in ns_l:
+        # Need to add "ns" to form bgpX so it is sent to the correct bgpX docker to handle the request
+        # If not MultiASIC, skip namespace argument
+        cmd = "show {} route {}".format(ipver, arg_strg)
+        if multi_asic.is_multi_asic():
+            output = bgp_util.run_bgp_command(cmd, ns)
+        else:
+            output = bgp_util.run_bgp_command(cmd)
+
+        # in case no output or something went wrong with user specified cmd argument(s) error it out
+        # error from FRR always start with character "%"
+        if output == "":
+            return
+        if output[0] == "%":
+            # remove the "json" keyword that was added by this handler to show original cmd user specified 
+            json_str = output[-5:-1]
+            if json_str == "json":
+                error_msg = output[:-5]
+            else:
+                error_msg = output
+            print error_msg
+            return
+        route_info = json.loads(output)
+        if filter_back_end or print_ns_str:
+            # clean up the dictionary to remove all the nexthops that are back-end interface
+            process_route_info(route_info, device, filter_back_end, print_ns_str, asic_cnt, ns, combined_route, back_end_intf_set)
+        else:
+            combined_route = route_info
+
+    if not found_json:
+        #print out the header if this is not a json request
+        if not filter_by_ip:
+            print_show_ip_route_hdr()
+        if print_ns_str:
+            for name_space, ns_route in sorted(combined_route.items()):
+                print "{}:".format(name_space)
```

Comment:
can you change all the print statement to be python3 compatible ?

Reply:
@arlakshm Sure. Will replace them to be python3 compatible.